lib/inspec/metadata.rb

```diff
@@ -177,27 +177,45 @@ def self.finalize_supports(supports, logger)
       end
     end
 
-    def self.finalize(metadata, profile_id, logger = nil)
+    def self.finalize_name(metadata, profile_id, original_target)
+      # profile_id always overwrites whatever already exists as the name
+      unless profile_id.to_s.empty?
+        metadata.params[:name] = profile_id.to_s
+        return
+      end
+
+      # don't overwrite an existing name
+      return unless metadata.params[:name].nil?
+
+      # if there's a title, there is no need to set a name too
+      return unless metadata.params[:title].nil?
```

Comment:
I am not sure if we should allow to set the name by `title`. This may be confusing. Do we have any additional value by supporting that? If the `inspec.yml` is available, but `name` is not specified, we could also display `(not specified)`